magellan-legacy/src/main/java/com/wealthfront/magellan/LegacyNavigator.kt

```diff
@@ -0,0 +1,54 @@
+package com.wealthfront.magellan
+
+import com.wealthfront.magellan.lifecycle.LifecycleAwareComponent
+import com.wealthfront.magellan.lifecycle.lifecycle
+import com.wealthfront.magellan.navigation.NavigableCompat
+import com.wealthfront.magellan.navigation.NavigationDelegate
+import com.wealthfront.magellan.navigation.NavigationEvent
+import com.wealthfront.magellan.navigation.Navigator
+import java.util.Stack
+
+class LegacyNavigator internal constructor(
+  container: () -> ScreenContainer
+) : Navigator, LifecycleAwareComponent() {
+
+  private val delegate by lifecycle(NavigationDelegate(container))
+
+  override val backStack: Stack<NavigationEvent>
+    get() = delegate.backStack
+
+  init {
+    delegate.currentNavigableSetup = { navItem ->
+      if (navItem is Screen<*>) {
+        navItem.setNavigator(this)
```

Comment:
Don't we need to do this when any screen is navigated to?

Reply:
I set this lambda to be executed when the screen is navigated to in the `NavigationDelegate`